### PR TITLE
Refactor(kcl-vet): Add method 'validate()' as the entry point of KCL-Vet.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #67

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm-tools/vet

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

add 'validator.rs' in 'kclvm/tools/src/vet'.
add method `validate()` as the entry point of KCL-Vet.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: kclvm/tools/src/vet/validator.rs [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+/// ```
+///
+/// The json file used above conforms to the schema rules, so the content of `result` you get is :
+/// ```ignore
+/// {
+/// "value": {
+///     "name": "Alice",
+///     "age": 18,
+///     "message": "This is Alice",
+///         "__settings__": {
```
Comment (Peefy) [inline kclvm/tools/src/vet/validator.rs:153]: remove `__settings__` it is a hidden attr.

File: kclvm/tools/src/vet/validator.rs [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+            pkgpath: String::new(),
+            ctx: ExprContext::Store,
+        })],
+        value: node,
+        type_annotation: None,
+        ty: None,
+    }))
+}
+
+const MAIN_PKG_NAME: &str = "__main__";
```
Comment (Peefy) [inline kclvm/tools/src/vet/validator.rs:252]: use `MAIN_PKG` in the kclvm_ast crate.

File: kclvm/tools/src/vet/validator.rs [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+///     "kcl_config_meta_file": "",
+///     "kcl_config_meta_line": 1,
+///     "kcl_config_meta_col": 1,
+///     "kcl_config_meta_arg_msg": "Instance check failed",
+///     "message": "",
+///     "err_type_code": 17,
+///     "is_warning": false
+/// }
+/// ```
+pub fn validate(
```
Comment (Peefy) [inline kclvm/tools/src/vet/validator.rs:191]: U can construct a ValidateOptions parameter.

File: kclvm/tools/src/vet/validator.rs [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+    schema_name: Option<String>,
+    attribute_name: &str,
+    validated_file_path: String,
+    validated_file_kind: LoaderKind,
+    kcl_path: Option<&str>,
+    kcl_code: Option<String>,
+) -> String {
```
Comment (Peefy) [inline kclvm/tools/src/vet/validator.rs:198]: Using `Result<String>`

File: kclvm/tools/src/lib.rs [hunk 5]
```diff
@@ -1,6 +1,6 @@
 pub mod format;
 pub mod lint;
-mod util;
+pub mod util;
```
Comment (Peefy) [inline kclvm/tools/src/lib.rs:3]: s/pub/pub(crate)

File: kclvm/tools/src/vet/expr_builder.rs [hunk 6]
```diff
@@ -135,7 +135,7 @@ impl ExprGenerator<serde_yaml::Value> for ExprBuilder {
 
                 for (k, v) in j_map.iter() {
                     let k = self
-                        .generate(k, schema_name)
+                        .generate(k, &None)
```
Comment (Peefy) [inline kclvm/tools/src/vet/expr_builder.rs:138]: U can leave a comment. `The configuration builder already in the schema no longer needs a schema name`

File: kclvm/tools/src/vet/test_datas/validate_cases/complex.k.stdout.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
+{
+    "value": {
+        "name": "Alice",
+        "age": 18,
+        "message": "This is Alice",
+        "data": {
+            "id": 1,
+            "value": "value1",
+            "__settings__": {
```
Comment (Peefy) [inline kclvm/tools/src/vet/test_datas/validate_cases/complex.k.stdout.json:9]: Omit the hidden private.

**Review by ldxdl — APPROVED** [review]
LGTM

File: kclvm/tools/src/vet/test_datas/validate_cases/complex.k.stdout.json [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        },
+        "labels": {
+            "key": "value"
+        },
+        "hc": [
+            1,
+            2,
+            3
+        ],
+        "__settings__": {
```
[the remark this answers is not in the document]
Comment (zong-zhe) [reply kclvm/tools/src/vet/test_datas/validate_cases/complex.k.stdout.json]: fixed

File: kclvm/tools/src/vet/test_datas/validate_cases/complex.k.stdout.json (hunk 7 above)
Comment (zong-zhe) [reply kclvm/tools/src/vet/test_datas/validate_cases/complex.k.stdout.json]: fixed

File: kclvm/tools/src/vet/expr_builder.rs (hunk 6 above)
Comment (zong-zhe) [reply kclvm/tools/src/vet/expr_builder.rs:139]: fixed

File: kclvm/tools/src/lib.rs (hunk 5 above)
Comment (zong-zhe) [reply kclvm/tools/src/lib.rs]: fixed

File: kclvm/tools/src/vet/validator.rs (hunk 4 above)
Comment (zong-zhe) [reply kclvm/tools/src/vet/validator.rs]: fixed

File: kclvm/tools/src/vet/validator.rs (hunk 3 above)
Comment (zong-zhe) [reply kclvm/tools/src/vet/validator.rs]: fixed

File: kclvm/tools/src/vet/validator.rs (hunk 2 above)
Comment (zong-zhe) [reply kclvm/tools/src/vet/validator.rs]: fixed

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**